pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -1464,10 +1466,28 @@ public void createSubscription(
                 throw new RestException(Response.Status.BAD_REQUEST, "Create subscription on non-persistent topic "
                         + "can only be done through client");
             }
-            MessageIdImpl messageId = resetCursorData == null ? null :
-                    new MessageIdImpl(resetCursorData.getLedgerId(), resetCursorData.getEntryId(),
-                            resetCursorData.getPartitionIndex());
-            internalCreateSubscription(asyncResponse, decode(encodedSubName), messageId, authoritative, replicated);
+            MessageIdImpl messageId = null;
+            Map<String, String> subscriptionProperties = null;
+            if (payload != null) {
+                if (payload.containsKey("properties")) {
```

Comment:
what about creating a class that extends ResetCursorData which contains the properties field? 
the code will look like this:

```
class CreateSubscriptionPayload extends ResetCursorData {
  Map<String, String> properties;
}

if (payload.containsKey("properties")) {
  ObjectMapperFactory.getThreadLocal()
                                        .convertValue(messageIdEncoded, CreateSubscriptionPayload.class);
} else {
  ObjectMapperFactory.getThreadLocal()
                                        .convertValue(messageIdEncoded, ResetCursorData.class);

}


```